amy/workshops/views.py

```diff
@@ -2186,11 +2187,22 @@ def get_initial(self, **kwargs):
 
         # Determine initial event in AwardForm
         if "find-training" in self.request.GET:
-            tasks = Person.objects.get(
-                pk=self.request.GET["person"]
-            ).get_training_tasks()
-            if tasks.count() == 1:
-                initial.update({"event": tasks[0].event})
+            initial.update(
+                {
+                    "badge": Badge.objects.get(name="instructor"),
+                }
+            )
+            progresses = TrainingProgress.objects.filter(
+                trainee__id=self.request.GET["person"],
+                requirement=TrainingRequirement.objects.get(name="Training"),
+                state="p",
+            )
+            if progresses.count() == 1:
+                initial.update(
+                    {
+                        "event": progresses[0].event,
+                    }
+                )
```

Comment:
We could use try-except here:
```suggestion
            try:
                progress = TrainingProgress.objects.get(
                    trainee__id=self.request.GET["person"],
                    requirement=TrainingRequirement.objects.get(name="Training"),
                    state="p",
                )
                initial["event"] = progress.event
            except (TrainingProgress.DoesNotExist, TrainingProgress.MultipleObjectsReturned):
                pass
```